lib/ecto/migration.ex

```diff
@@ -801,6 +801,14 @@ defmodule Ecto.Migration do
     validate_type!(subtype)
   end
 
+  defp validate_type!({type, subtype}) when is_atom(type) and is_list(subtype) do
+    for t <- subtype, do: validate_type!(t)
+  end
+
+  defp validate_type!({type, subtype}) when is_atom(type) and is_tuple(subtype) do
```

Comment:
That's a list though? And is that always 3 elements or it may be 2, 3, 4 and more?